lib/tumblr.js

```diff
@@ -1,262 +1,452 @@
 
-  // Helpers
+        return request.call(client, buildApiPath(apiPathArgs), params, callback);
+    };
 
-  _get: function (path, params, callback, addApiKey) {
+    set(client, methodName, createFunction(methodName, namedParams, methodBody));
+}
 
-    params = params || {};
-    if (addApiKey) {
-      params.api_key = this.credentials.consumer_key;
+/**
+ * ## adtMethods
+ *
+ * Adds methods to the client
+ *
+ * @param  {TumblrClient} client: an instance of the `tumblr.js` API client
+ * @param  {Object} methods: mapping of method names to endpoints. Endpoints can be a string or an
+ *         array of format `[apiPathString, requireParamsArray]`
+ * @param  {String|Function} requestType: the request type or a function that makes the request
+ */
+function addMethods(client, methods, requestType) {
+    var apiPath, paramNames;
+    for (var methodName in methods) {
+        apiPath = methods[methodName];
+        if (typeof apiPath === 'string') {
+            paramNames = [];
+        } else if (isPlainObject(apiPath)) {
+            paramNames = apiPath.paramNames || [];
+            apiPath = apiPath.path;
+        } else {
+            paramNames = apiPath[1] || [];
+            apiPath = apiPath[0];
+        }
+        addMethod(client, methodName, apiPath, paramNames, requestType || 'GET');
     }
+}
 
-    request.get({
-      url: baseURL + path + '?' + qs.stringify(params),
-      json: true,
-      oauth: this.credentials,
-      followRedirect: false,
-      headers : {
-        'User-Agent' : 'tumblr.js/' + version
-      }
-    }, requestCallback(callback));
-
-  },
-
-  _post: function (path, params, callback) {
-
-    var data = params.data;
-    delete params.data;
+/**
+ * ## wrapCreatePost
+ *
+ * Wraps createPost to specify `type` and validate the parameters
+ *
+ * @param  {String} type: post type
+ * @param  {Function} [validate]: returns `true` if the parameters validate
+ *
+ * @return {Function} wrapped function
+ */
+function wrapCreatePost(type, validate) {
+    return function(blogIdentifier, params, callback) {
+        params = extend({type: type}, params);
+
+        if (isArray(validate)) {
+            validate = partial(function(params, requireKeys) {
+                if (requireKeys.length) {
+                    var keyIntersection = intersection(keys(params), requireKeys);
+                    if (requireKeys.length === 1 && !keyIntersection.length) {
+                        throw new Error('Missing required field: ' + requireKeys[0]);
+                    } else if (!keyIntersection.length) {
+                        throw new Error('Missing one of: ' + requireKeys.join(', '));
+                    } else if (keyIntersection.length > 1) {
+                        throw new Error('Can only use one of: ' + requireKeys.join(', '));
+                    }
+                }
+                return true;
+            }, params, validate);
+        }
+
+        if (typeof validate === 'function') {
```

Comment:
I've seen a handful of `typeof ... === 'function'`; is it worthwhile to require `isFunction` from lodash and use it like you're doing with `isArray`?